improver/utilities/mathematical_operations.py

```diff
@@ -638,14 +651,309 @@ def process(
+                )
+        elif not spatial_coords_match(cubes_to_check):
+            raise ValueError("The spatial coordinates must match.")
+
+    def verify_time_coords_match(
+        self, anomaly_cube: Cube, mean_cube: Cube, std_cube: Optional[Cube] = None
+    ) -> None:
+        """Check that all cubes have compatible time coordinates.
+
+        Raises:
+            ValueError: If the time coordinates of the cubes are incompatible.
+        """
+        errors = []
+
+        # Check if mean and standard deviation cubes have the same bounds
+        if std_cube and not np.array_equal(
+            mean_cube.coord("time").bounds, std_cube.coord("time").bounds
+        ):
+            errors.append(
+                "The mean and standard deviation cubes must have compatible bounds. "
+                "The following bounds were found: "
+                f"mean_cube bounds: {mean_cube.coord('time').bounds},"
+                f"std_cube bounds: {std_cube.coord('time').bounds}"
+            )
+
+        # Check if anomaly cube's time points fall within the bounds of the
+        # mean cube.
+        # The verification of the standard deviation cube's bounds suitably matching the
+        # anomaly cube's is covered implicitly due to the above code chunk.
+        if not self.ignore_temporal_mismatch:
+            if anomaly_cube.coord("reference_epoch") is not None:
+                if np.any(
+                    anomaly_cube.coord("reference_epoch").points
+                    != mean_cube.coord("time").points
+                ) or not np.all(
+                    anomaly_cube.coord("reference_epoch").bounds
+                    == mean_cube.coord("time").bounds
+                ):
+                    errors.append(
+                        "The reference epoch coordinate of the anomaly cube must "
+                        "match the time coordinate of the mean cube. I.e. The same mean"
+                        "cube that was used to calculate the anomaly."
+                        "The following values were found: "
+                        f"anomaly cube reference epoch: {anomaly_cube.coord('reference_epoch').points},"
+                        f"mean cube time: {mean_cube.coord('time').points},"
+                        f"anomaly cube reference epoch bounds: {anomaly_cube.coord('reference_epoch').bounds},"
+                        f"mean cube time bounds: {mean_cube.coord('time').bounds}"
+                    )
+
+        if errors:
+            raise ValueError("\n".join(errors))
+
+    @staticmethod
+    def calculate_forecast_value(
+        anomaly_cube: Cube, mean_cube: Cube, std_cube: Optional[Cube] = None
+    ) -> ndarray:
+        """Calculate forecast value(s) from the input cubes."""
+        forecast_value = anomaly_cube.data + mean_cube.data
+        if std_cube:
+            forecast_value = (anomaly_cube.data * std_cube.data) + mean_cube.data
+        return forecast_value
```

Comment:
```suggestion
    @staticmethod
    def calculate_forecast_value(
        anomaly_cube: Cube, mean_cube: Cube, std_cube: Optional[Cube] = None
    ) -> ndarray:
        """Calculate forecast value(s) from the input cubes."""
        if std_cube:
            forecast_value = (anomaly_cube.data * std_cube.data) + mean_cube.data
        else:
            forecast_value = anomaly_cube.data + mean_cube.data
            
        return forecast_value
```

There's no need to do the calculation twice.